tests/test_core.py

```
@@ -114,4 +115,113 @@ def test_is_valid_api_key(mock_settings):
+
+    def unexpected_read(_: object) -> dict[str, int]:
+        pytest.fail("contains reloaded whitelist without a storage change")
+
+    monkeypatch.setattr(core, "_read_whitelist_file", unexpected_read)
+
+    assert store.contains(address, now=int(time.time())) is False
+    assert store.contains(address, now=int(time.time())) is False
+
+
+@pytest.mark.parametrize(
+    "helper, expected_message",
```

Ruff (0.15.13)

[warning] Wrong type passed to first argument of `pytest.mark.parametrize`; expected `tuple`

Use a `tuple` for the first argument

(PT006)